Remove SIRF_INSTALL_PATH and SIRF_PATH environment variables in `example_data_path` (#1412)
SyneRBI/SIRF: examples/README.md

The contents of subfolder named Python are described [here](Python/README.md). The contents of subfolder named Matlab are of similar nature but are not documented as we no longer support Matlab.

## Environment variables

The correct setup of SIRF requires defining the environment variables `SIRF_PATH` and `SIRF_INSTALL_PATH`.

If you built SIRF with the SIRF-Superbuild, these environment variables can be sourced from the `env_sirf.*` files, e.g.
on Unix/Linux [`env_sirf.sh`](https://github.com/SyneRBI/SIRF-SuperBuild/blob/master/env_sirf.sh.in).

Alternatively,  `SIRF_PATH` should be set by the user to the full path to the root folder containing the SIRF source,
and `SIRF_INSTALL_PATH` to the full path to the folder where the user installed SIRF binaries and other files (i.e.
set by `CMAKE_INSTALL_PREFIX`).

Additionally, the user may define the variable `SIRF_DATA_PATH` to point to an alternative location. For instance, when the
installation files are in a write-protected location, the user can make a copy elsewhere and point there.

<!--
(NOTE: even on Windows, you must use `/` in paths, not `\\`.)
-->

> 

Reviewer: ## Environment variables

The user may define the environment variable `SIRF_DATA_PATH` to override the install location. For instance, when the
installation files are in a write-protected location, the user can make a copy elsewhere and point there, as
the demos might write in that location.

Reviewer: ok, I see now that this is covered below, so agree to remove this.

> 3. The directory `${CMAKE_INSTALL_PREFIX}/share/SIRF-<version_major>.<version_minor>`

Reviewer: As far as I can see, option 2 is not implemented (which is fine by me). If you do implement it, then you need to correct the various spelling mistakes here. Otherwise, simply change
```suggestion
1. The directory pointed by `${SIRF_DATA_PATH}/examples` if the environment variable `SIRF_DATA_PATH` is set by the user.
2. The directory `${CMAKE_INSTALL_PREFIX}/share/SIRF-<version_major>.<version_minor>`
```

Author: > As far as I can see, option 2 is not implemented

This option can be taken by the user by defining `SIRF_EXAMPLES_DATA_PATH` when calling `cmake` (cf. lines 71-73 in `SIRF/CMakeLists.txt`).

Reviewer: ok, but this isn't an environment variable, but a CMake variable, so I suppose we get

```suggestion
1. The directory pointed by `${SIRF_DATA_PATH}/examples` if the environment variable `SIRF_DATA_PATH` is set by the user.
2. The directory pointed by the CMake variable `SIRF_EXAMPLES_DATA_PATH` if it is set by the user during building SIRF.
3. The directory `${CMAKE_INSTALL_PREFIX}/share/SIRF-<version_major>.<version_minor>`
```

The contents of subfolder named Python are described [here](Python/README.md). The contents of subfolder named Matlab are of similar nature but are not documented as we no longer support Matlab.

### Examples data path

The resolution order for the examples data path is:

1. The directory pointed by `${SIRF_DATA_PATH}/examples` if the environment variable `SIRF_DATA_PATH` is set by the user.
2. The directory pointed by the CMake variable `SIRF_EXAMPLES_DATA_PATH` if it is set by the user during building SIRF.
3. The directory `${CMAKE_INSTALL_PREFIX}/share/SIRF-<version_major>.<version_minor>`

> The user may need options 1 or 2 e.g. if the examples input data is in a write-protected location, and the examples are designed to send output there, hence a copy to writable location is needed.

Reviewer: ```suggestion

The user may need options 1 or 2 e.g. if the examples input data is in a write-protected location, and the examples are designed to send output there, hence a copy to writable location is needed.
```
